### PR TITLE
Fix for test_pvc_delete_verify_size_is_returned_to_backendpool

### PR DESCRIPTION
**Pull request by parikshithb**
Issue #643

### CODE REVIEW

**Comment by parikshithb** [comment]
Log : http://pastebin.test.redhat.com/791743

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 1]
```diff
@@ -44,34 +45,39 @@ def check_ceph_used_space():
...
+    logger.info("Verifying whether PV is deleted")
     try:
-        PV.get(pvc_obj.backed_pv)
-    except CommandFailed as ecf:
+        assert helpers.validate_pv_delete(pvc_obj.backed_pv)
+    except AssertionError as ecf:
         assert "not found" in str(ecf), (
-            f"Unexpected: pv {pvc_obj.backed_pv} still exists"
-        )
+             f"Unexpected: pv {pvc_obj.backed_pv} still exists"
```
Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: delete extra indentations

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 2]
```diff
@@ -44,34 +45,39 @@ def check_ceph_used_space():
...
     logger.info(f"Verifying pv {pvc_obj.backed_pv} exists on backend")
 
-    _rc = pvc_obj.verify_pv_exists_in_backend(cbp_name)
+    _rc = helpers.verify_volume_deleted_in_backend(
+        interface=constants.CEPHBLOCKPOOL, image_uuid=rbd_image_id,
+        pool_name=cbp_name
+    )
 
-    if _rc:
+    if _rc is False:
```
Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: if not _rc

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 3]
```diff
@@ -44,34 +45,39 @@ def check_ceph_used_space():
...
         raise UnexpectedBehaviour(f"pv {pvc_obj.backed_pv} exists on backend")
     logger.info(
-        f"Expected: pv {pvc_obj.backed_pv} doesn't exist on backend after deleting pvc"
+        f"Expected: pv {pvc_obj.backed_pv} "
```
Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: Change pv to PV

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 4]
```diff
@@ -44,34 +45,39 @@ def check_ceph_used_space():
...
+        f"doesn't exist on backend after deleting pvc"
```
Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: change pvc to PVC

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 5]
```diff
@@ -44,34 +45,39 @@ def check_ceph_used_space():
...
     used_space = used_in_gb
     raise UnexpectedBehaviour(
-        f"In Ceph status, used size is keeping varying"
+        f"In Ceph status, used size is varying"
     )
 
 
 @retry(UnexpectedBehaviour, tries=5, delay=3, backoff=1)
-def verify_pv_not_exists(pvc_obj, cbp_name):
+def verify_pv_not_exists(pvc_obj, cbp_name, rbd_image_id):
```
Comment (ebenahar) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: This function seems redundant. 
Would it make sense to call helpers.verify_volume_deleted_in_backend from within helpers.validate_pv_delete()?

Comment (ebenahar) [reply tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: @parikshithb Please proceed with the current implementation, we can do this change later as part of a separate PR

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 6]
```diff
@@ -88,8 +94,11 @@ def create_pvc_and_verify_pvc_exists(sc_name, cbp_name):
     pvc_obj.reload()
 
     # Validate pv is created on ceph
-    logger.info(f"Verifying pv exists on backend")
-    pvc_obj.verify_pv_exists_in_backend(cbp_name)
+    logger.info(f"Verifying PV exists on backend")
+    assert helpers.verify_volume_deleted_in_backend(
+        interface=constants.CEPHBLOCKPOOL, image_uuid=pvc_obj.image_uuid,
+        pool_name=cbp_name
+    ) is False
```
Comment (RazTamir) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: please do 
```
assert not helpers.verify_volume_deleted_in_backend(...)
```

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 7]
```diff
@@ -122,13 +132,22 @@ def test_pvc_delete_and_verify_size_is_returned_to_backend_pool(self):
...
-        verify_pv_not_exists(pvc_obj, self.cbp_obj.name)
+        rbd_image_id = pvc_obj.image_uuid
+        for resource in pod_obj, pvc_obj:
+            resource.delete()
+            try:
+                resource.ocp.wait_for_delete(resource)
+            except TimeoutError:
+                failed_to_delete.append(resource)
+        if failed_to_delete:
+            raise ResourceLeftoversException(
```
Comment (RazTamir) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: ResourceLeftoversException is for the environment check leftovers.
Please use Unexpected behavior exception

Comment (parikshithb) [reply tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: ack

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py (hunk 6 above)
Comment (parikshithb) [reply tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: ack

File: tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py [hunk 8]
```diff
@@ -44,37 +44,42 @@ def check_ceph_used_space():
...
     """
     Ensure that pv does not exists
     """
 
     # Validate on ceph side
-    logger.info(f"Verifying pv {pvc_obj.backed_pv} exists on backend")
+    logger.info(f"Verifying PV {pvc_obj.backed_pv} exists on backend")
 
-    _rc = pvc_obj.verify_pv_exists_in_backend(cbp_name)
+    _rc = helpers.verify_volume_deleted_in_backend(
```
Comment (RazTamir) [inline tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: usually rc (return code) implies if the function failed or succeeded with rc>=0 (failure) and rc=0 success.
This usage of _rc is confusing.
Could you please change the parameter name to 'status' or 'deleted'?

Comment (parikshithb) [reply tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py]: ack

**Comment by parikshithb** [comment]
Verified. Jenkins [log](http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/pbyregow-ocs-cluster/pbyregow-ocs-cluster_20190905T175934/logs/ocs-ci-logs-1567709012/tests/manage/pv_services/test_pvc_delete_verify_size_is_returned_to_backendpool.py/TestPVCDeleteAndVerifySizeIsReturnedToBackendPool/test_pvc_delete_and_verify_size_is_returned_to_backend_pool/logs)

**Pull request merged**